flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/AggregatedStatisticsTracker.java

```diff
@@ -30,104 +42,225 @@
  * {@link AggregatedStatistics} received from {@link DataStatisticsOperator} subtasks for specific
  * checkpoint.
  */
-class AggregatedStatisticsTracker<D extends DataStatistics<D, S>, S> {
+class AggregatedStatisticsTracker {
   private static final Logger LOG = LoggerFactory.getLogger(AggregatedStatisticsTracker.class);
-  private static final double ACCEPT_PARTIAL_AGGR_THRESHOLD = 90;
+
   private final String operatorName;
-  private final TypeSerializer<DataStatistics<D, S>> statisticsSerializer;
   private final int parallelism;
-  private final Set<Integer> inProgressSubtaskSet;
-  private volatile AggregatedStatistics<D, S> inProgressStatistics;
+  private final TypeSerializer<DataStatistics> statisticsSerializer;
+  private final int downstreamParallelism;
+  private final StatisticsType statisticsType;
+  private final int switchToSketchThreshold;
+  private final Comparator<StructLike> comparator;
+  private final NavigableMap<Long, Aggregation> aggregationsPerCheckpoint;
+
+  private volatile AggregatedStatistics completedStatistics;
```

Comment:
How is the thread model work for the event handling? Do we need the `volatile` here?

Reply:
good question. we don't really need `volatile` here as coordinator event handling is always single thread. let me remove the volatile.

```
this.coordinatorExecutor = Executors.newSingleThreadExecutor(coordinatorThreadFactory);
```